src/geometry/manifold/basic_smooth_bundle.lean

```diff
@@ -0,0 +1,570 @@
+topological_fiber_bundle_core.continuous_proj _
+
+/-- The tangent bundle projection on the basis is an open map. -/
+lemma tangent_bundle_proj_open : is_open_map (tangent_bundle.proj I M) :=
+topological_fiber_bundle_core.is_open_map_proj _
+
+/-- In the tangent bundle to the model space, the charts are just the identity-/
+@[simp] lemma tangent_bundle_model_space_chart_at (p : tangent_bundle I H) :
+  (chart_at (H × E) p).to_local_equiv = local_equiv.refl (H × E) :=
+begin
+  have A : ∀x_fst, fderiv_within 𝕜 (I.to_fun ∘ I.inv_fun) (range I.to_fun) (I.to_fun x_fst)
+           = continuous_linear_map.id,
+  { assume x_fst,
+    have : fderiv_within 𝕜 (I.to_fun ∘ I.inv_fun) (range I.to_fun) (I.to_fun x_fst)
+         = fderiv_within 𝕜 id (range I.to_fun) (I.to_fun x_fst),
+    { refine fderiv_within_congr (I.unique_diff _ (mem_range_self _)) (λy hy, _) (by simp),
+      exact model_with_corners_right_inv _ hy },
+    rwa fderiv_within_id (I.unique_diff _ (mem_range_self _)) at this },
+  ext x : 1,
+  show (chart_at (H × E) p).to_fun x = (local_equiv.refl (H × E)).to_fun x,
+  { cases x,
+    simp [chart_at, manifold.chart_at, basic_smooth_bundle_core.chart,
+          topological_fiber_bundle_core.local_triv, topological_fiber_bundle_core.local_triv',
+          basic_smooth_bundle_core.to_topological_fiber_bundle_core, tangent_bundle_core],
+    erw [local_equiv.refl_to_fun, local_equiv.refl_inv_fun, A],
+    refl },
+  show ∀x, ((chart_at (H × E) p).to_local_equiv).inv_fun x = (local_equiv.refl (H × E)).inv_fun x,
+  { rintros ⟨x_fst, x_snd⟩,
+    simp [chart_at, manifold.chart_at, basic_smooth_bundle_core.chart,
+          topological_fiber_bundle_core.local_triv, topological_fiber_bundle_core.local_triv',
+          basic_smooth_bundle_core.to_topological_fiber_bundle_core, tangent_bundle_core],
+    erw [local_equiv.refl_to_fun, local_equiv.refl_inv_fun, A],
+    refl },
+  show ((chart_at (H × E) p).to_local_equiv).source = (local_equiv.refl (H × E)).source,
+  by simp [chart_at, manifold.chart_at, basic_smooth_bundle_core.chart,
+           topological_fiber_bundle_core.local_triv, topological_fiber_bundle_core.local_triv',
+           basic_smooth_bundle_core.to_topological_fiber_bundle_core, tangent_bundle_core,
+           local_equiv.trans_source]
+end
+
+variable (H)
+/-- In the tangent bundle to the model space, the topology is the product topology, i.e., the bundle
+is trivial -/
+lemma tangent_bundle_model_space_topology_eq_prod :
+  tangent_bundle.topological_space I H = prod.topological_space :=
+begin
+  ext o,
+  let x : tangent_bundle I H := (I.inv_fun (0 : E), (0 : E)),
+  let e := chart_at (H × E) x,
+  have e_source : e.source = univ, by { simp [e], refl },
+  have e_target : e.target = univ, by { simp [e], refl },
+  let e' := e.to_homeomorph_of_source_eq_univ_target_eq_univ e_source e_target,
+  split,
+  { assume ho,
+    have := e'.continuous_inv_fun o ho,
+    simp [e', tangent_bundle_model_space_chart_at] at this,
```

Comment:
Is there a reason to avoid
```suggestion
    simpa [e', tangent_bundle_model_space_chart_at] using this
```
here and below?

Reply:
I remember some cases where this didn't work. But here it works fine (and thanks to the `simpa` magic syntax you can even remove the `using this`). Done.